script/install-local-registry.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+CONTROL_PLANE_CONTAINER="kubeapps-ci-control-plane"
+REGISTRY_NS=ci
+DOCKER_REGISTRY_HOST=local-docker-registry
+DOCKER_REGISTRY_PORT=5600
+DOCKER_REGISTRY_URL="https://$DOCKER_REGISTRY_HOST:$DOCKER_REGISTRY_PORT"
+DOCKER_USERNAME=testuser
+DOCKER_PASSWORD=testpassword
+
+installLocalRegistry() {
+    PROJECT_PATH=$1
+
+    kubectl create namespace $REGISTRY_NS
+
+    # Add our CA to the node
+    docker cp "$(mkcert -CAROOT)"/rootCA.pem "$CONTROL_PLANE_CONTAINER:/usr/share/ca-certificates/local-rootCA.pem"
+    docker exec --user root $CONTROL_PLANE_CONTAINER sh -c "echo 'local-rootCA.pem' >> /etc/ca-certificates.conf"
+    docker exec --user root $CONTROL_PLANE_CONTAINER update-ca-certificates -f
+
+    # Restart containerd to make the CA addition effective
+    docker exec --user root $CONTROL_PLANE_CONTAINER systemctl restart containerd
+
+    # Generate new certificate for registry
+    mkcert -key-file $PROJECT_PATH/devel/localhost-key.pem -cert-file $PROJECT_PATH/devel/docker-registry-cert.pem $DOCKER_REGISTRY_HOST "docker-registry.$REGISTRY_NS.svc.cluster.local"
+    kubectl -n $REGISTRY_NS delete secret registry-tls --ignore-not-found=true
+    kubectl -n $REGISTRY_NS create secret tls registry-tls --key $PROJECT_PATH/devel/localhost-key.pem --cert $PROJECT_PATH/devel/docker-registry-cert.pem
+
+    # Create registry resources
+    kubectl apply -f ${PROJECT_PATH}/integration/registry/local-registry.yaml
+
+    # Wait for deployment to be ready
+    kubectl rollout status -w deployment/private-registry -n $REGISTRY_NS
+
+    # Add registry to node hosts.
+    # Haven't found a way to use the cluster DNS from the node, 
+    # following https://github.com/kubernetes/kubernetes/issues/8735#issuecomment-148800699
+    REGISTRY_IP=$(kubectl -n $REGISTRY_NS get service/docker-registry -o jsonpath='{.spec.clusterIP}')
+    docker exec --user root $CONTROL_PLANE_CONTAINER sh -c "echo '$REGISTRY_IP  $DOCKER_REGISTRY_HOST' >> /etc/hosts"
+}
+
+pushContainerToLocalRegistry() {
+    DOCKER_REGISTRY="$DOCKER_REGISTRY_HOST:$DOCKER_REGISTRY_PORT"
+
+    echo "127.0.0.1  $DOCKER_REGISTRY_HOST" | sudo tee -a /etc/hosts
+
+    docker pull nginx
+    docker tag nginx $DOCKER_REGISTRY/nginx
+
+    /bin/sh -c "kubectl -n ${REGISTRY_NS} port-forward service/docker-registry 5600:5600 &"
```

Comment:
Solution far from elegant, but it does the trick for pushing from CI host into Docker registry inside cluster.

Reply:
setting up a port-forward temporarily is OK I think, we do the same for chartmuseum. But the sleeps are more worrying in my opinion (as the time required is variable). It'd be nice if we could instead wait until the service is available (with a simple curl or something).